### PR TITLE
[Bugfix:Forum] Invalid Attachment

### PR DESCRIPTION
**Pull request by jportorreal00**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Fixes [#5390](https://github.com/Submitty/Submitty/issues/5390)- When a user tries to upload an attachment that is not an image, a warning is given. If a user decides to ignore the error and submit a post with an invalid attachment, the submit button is frozen.

### What is the new behavior?
Any invalid attachment gives a warning but is automatically removed

### Other information?
I tested this by trying to post threads and posts with attachments. I added image files, non-image files, or a combination of both.

### CODE REVIEW

File: site/public/js/forum.js [hunk 1]
```diff
@@ -34,12 +34,15 @@ function openFileForum(directory, file, path ){
...
 
-function checkForumFileExtensions(files){
-    var count = 0;
+function checkForumFileExtensions(post_box_id, files){
+    var count = files.length;
     for(var i = 0; i < files.length; i++){
         var extension = getFileExtension(files[i].name);
-        if(extension == "gif" || extension == "png" || extension == "jpg" || extension == "jpeg" || extension == "bmp"){
-            count++;
+        if( !(extension == "gif" || extension == "png" || extension == "jpg" || extension == "jpeg" || extension == "bmp") ){
```
Comment (MasterOdin) [inline site/public/js/forum.js:41]:
```suggestion
        if(!['gif', 'png', 'jpg', 'jpeg', 'bmp'].includes(extension)) {
```

File: site/public/js/forum.js [hunk 2]
```diff
@@ -34,12 +34,15 @@ function openFileForum(directory, file, path ){
...
+            deleteSingleFile(files[i].name, post_box_id, false);
+            removeLabel(files[i].name, post_box_id);
+            files.splice(i, 1);
+            i--;
         }
     } return count == files.length;
```
Comment (MasterOdin) [inline site/public/js/forum.js:47]:
```suggestion
    }
    return count == files.length;
```

File: site/public/js/forum.js [hunk 3]
```diff
@@ -34,12 +34,15 @@ function openFileForum(directory, file, path ){
     window.open(url,"_blank","toolbar=no,scrollbars=yes,resizable=yes, width=700, height=600");
 }
 
-function checkForumFileExtensions(files){
-    var count = 0;
+function checkForumFileExtensions(post_box_id, files){
+    var count = files.length;
```
Comment (MasterOdin) [inline site/public/js/forum.js:38]:
```suggestion
    let count = files.length;
```

File: site/public/js/forum.js [hunk 4]
```diff
@@ -34,12 +34,15 @@ function openFileForum(directory, file, path ){
...
     for(var i = 0; i < files.length; i++){
         var extension = getFileExtension(files[i].name);
```
Comment (MasterOdin) [inline site/public/js/forum.js:40]:
```suggestion
    for (let i = 0; i < files.length; i++) {
        let extension = getFileExtension(files[i].name);
```

File: site/public/js/forum.js (hunk 1 above)
Comment (jportorreal00) [reply site/public/js/forum.js]: I've combined all of the changes into a commit :)

**Comment by marwanatef2** [comment]
@jportorreal00 if you made new changes upon Matts request please **"resolve conversation"** so not to add any suggestion twice.

**Comment by jportorreal00** [comment]
@marwanatef2 Got it! Thanks for the advice

**Comment by jportorreal00** [comment]
@marwanatef2 I see. While testing I realized that posting a thread does not disable the submit button, only posts. Thanks for the help!

File: site/app/templates/forum/GeneratePostList.twig [hunk 5]
```diff
@@ -51,7 +51,7 @@
 
     <hr id="post-hr" />
 
-    <form class="post_reply_from" method="POST" onsubmit="post.disabled=true; post.value='Submitting post...'; return true;" action="{{ form_action_link }}" enctype="multipart/form-data">
+    <form class="post_reply_from" method="POST" return true;" action="{{ form_action_link }}" enctype="multipart/form-data">
```
Comment (marwanatef2) [inline site/app/templates/forum/GeneratePostList.twig:54]:
```suggestion
    <form class="post_reply_from" method="POST" action="{{ form_action_link }}" enctype="multipart/form-data">
```

File: site/public/js/forum.js [hunk 6]
```diff
@@ -1358,8 +1358,6 @@ function thread_post_handler(){
     $('.post_reply_from').submit(function(){
```
Comment (marwanatef2) [inline site/public/js/forum.js:1358]: this whole function is for disabling the submit button, the first two lines select the elements while the next two lines apply the disabling so it should be removed (lines 1358 : 1362)

**Pull request merged**